src/PerpetualPositionRouter.sol

```diff
@@ -139,12 +149,23 @@ contract PerpetualPositionRouter {
   //
   // 1. What us a reasonable amount of precision to reduce the function?
   // 2. What are reasonable time periods for deadlines
-  //
-  // closePosition is not using the amount value and maybe we could optimize by splitting it into a
-  // separate contract
   fallback() external payable {
-    (uint8 funcName, uint256 amount, uint256 oppositeAmountBound, uint160 sqrtPriceLimitX96) =
-      abi.decode(msg.data, (uint8, uint256, uint256, uint160));
+    // The first 11 bytes are padding.
+    // The 12th byte will contain the function name
+    // which we will use to pick the correct decoding
+    // strategy
```

Comment:
I think the idea is that `msg.data` is going to have to be 3 words long, since `amount` and `oppositeAmoutBound` are both `uint256s`. This leaves 256 bits for `combinedArgs` and `funcName`, and 256-(168 + 8) = 80 bits remaining.

But that's only 10 bytes of padding, not 11. Am I confused here?

Reply:
Everything is right except funcName is a part of combinedArgs meaning it will only be 256 - 168 leaving 88 bits remaining.

And from my understanding the padding is there because we are using abi.encode to create the calldata which forces every field to be 32 bytes. Since this is the first argument we could create a 21 byte hex string and append it to the abi encoded data to get even more savings.

Reply:
> And from my understanding the padding is there because we are using abi.encode to create the calldata which forces every field to be 32 bytes.

yea that is correct, to remove those zero bytes you'd have to manually decode the calldata instead of `abi.decode`. That's something for a follow up PR but it shouldn't be too hard

Reply:
https://github.com/ScopeLift/optimizoors-perpetual-protocol/issues/16#issue-1714841796